sycl/include/sycl/platform.hpp

```diff
@@ -194,6 +194,11 @@ class __SYCL_EXPORT platform : public detail::OwnerLessBase<platform> {
   /// \return a vector of all available SYCL platforms.
   static std::vector<platform> get_platforms();
 
+  /// Returns all unsupported (non-SYCL) platforms in the system.
+  ///
+  /// \return a vector of all unsupported non-SYCL platforms.
+  static std::vector<platform> get_unsupported_platforms();
```

Comment:
@jchlanda - Sorry for the post-commit review, but I think we need to reconsider the design of this. This function is now a public interface of a SYCL class, but is not part of neither the standard nor an extension. We either need an extension written for this, or we need it hidden, either in the `detail` namespace, by making it `private` or both.